test/de/schildbach/pte/live/TpgProviderLiveTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright the original author or authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package de.schildbach.pte.live;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Date;
+
+import org.junit.Test;
+
+import de.schildbach.pte.TpgProvider;
+import de.schildbach.pte.dto.Location;
+import de.schildbach.pte.dto.LocationType;
+import de.schildbach.pte.dto.NearbyLocationsResult;
+import de.schildbach.pte.dto.Point;
+import de.schildbach.pte.dto.QueryDeparturesResult;
+import de.schildbach.pte.dto.QueryTripsResult;
+import de.schildbach.pte.dto.SuggestLocationsResult;
+
+/**
+ * @author Andreas Schildbach
+ */
+public class TpgProviderLiveTest extends AbstractProviderLiveTest {
+    public TpgProviderLiveTest() {
+        super(new TpgProvider(secretProperty("tpg.api_authorization")));
+    }
+
+    @Test
+    public void nearbyStationsByCoordinate() throws Exception {
+        final NearbyLocationsResult result = queryNearbyStations(Location.coord(46209460, 6167140)); // Parc et plage des Eaux-Vives
+        print(result);
+    }
+
+    @Test
+    public void queryDepartures() throws Exception {
+        final QueryDeparturesResult result = queryDepartures("8501238", false); // Genève-Paquis
+        print(result);
+    }
+
+    @Test
+    public void queryDeparturesSuburbanTrain() throws Exception {
+        final QueryDeparturesResult result = queryDepartures("8516155", false); // Lancy-Pont-Rouge
+        print(result);
+    }
+
+    @Test
+    public void queryDeparturesTram() throws Exception {
+        final QueryDeparturesResult result = queryDepartures("8587907", true); // Plainpalais
+        print(result);
+    }
+
+    // @Test
+    // public void queryDeparturesTrolley() throws Exception {
+    //     final QueryDeparturesResult result = queryDepartures("8591177", false); // Hardplatz
+    //     print(result);
+    // }
```

Comment:
How can I know which one are Trolley?

Reply:
It's fine if you provide one generic `queryDepartures()` test. The ones you copied and pasted are somehow special, but I currently don't remember details.